service-app/internal/factory/document_processor.go

```diff
@@ -0,0 +1,31 @@
+package factory
+
+import (
+	"fmt"
+
+	"github.com/ministryofjustice/opg-scanning/internal/types"
+)
+
+type DocumentProcessor struct {
+	doc       interface{}
+	validator types.Validator
+	sanitizer types.Sanitizer
+}
+
+func (p *DocumentProcessor) Process() (interface{}, error) {
+	if err := p.validator.Validate(); err != nil {
+		return nil, fmt.Errorf("validation failed: %w", err)
```

Comment:
If validation fails we don't actually want to reject the document: when Sirius does ingestion and runs over these rules it will continue but mark them as issues for someone to manually review.

Reply:
Ok, well noted, I will need to probably refactor some parts to make a more robust solution. I don't fully understand whats involved in marking them and how this its communicated downstream. For now though I put a TODO, and logged the error(s) without blocking the processing.